extensions/kafka/src/main/java/com/hazelcast/jet/kafka/impl/WriteKafkaP.java

```diff
@@ -181,8 +181,7 @@ private void recoverTransaction(KafkaTransactionId txnId, boolean commit) {
         properties2.put("transactional.id", txnId.getKafkaId());
         try (KafkaProducer<?, ?> p  = new KafkaProducer<>(properties2)) {
             if (commit) {
-                ResumeTransactionUtil.resumeTransaction(p, txnId.producerId(), txnId.epoch(),
-                        txnId.getKafkaId());
+                ResumeTransactionUtil.resumeTransaction(p, txnId.producerId(), txnId.epoch());
                 try {
                     p.commitTransaction();
```

Comment:
Not related to this line, but to this file: We must prevent the abortion of transactions when the producer is closed, we must add `Duration.ZERO` to `KafkaTransaction.release()`. For some reason the test doesn't catch this failure, but I'm pretty sure it must be there. I was unable to fix the test to catch this, if you manage to do this, it would be great.